File: basic_prob/reverse_array.py
def reverse_array(a, start, end):
	while start >= end:
		return
	temp = a[start]
	a[start] = a[end]
	a[end] = temp

	reverse_array(a, start+1, end-1)
def print_array(a, n):
	for i in range(n):
		print(a[i], end = " ")



a = [1,2,3,4,6,7]
print_array(a, 6)
reverse_array(a, 0, 5)
print("\n")
print_array(a, 6)

File: basic_prob/Reverse_array.cpp
#include<iostream>
using namespace std;

void reverse_array(int a[], int start, int end)
{
	while(start >= end)
		return;
	int temp = a[start];
	a[start] = a[end];
	a[end] = temp;

	reverse_array(a, start+1, end-1);
}
void print_array(int a[], int n)
{
	for(int i = 0; i<n;i++)
	{
		cout<<a[i]<<" ";
	}
	cout<<endl;
}

int main()
{
	int a[] = {1,3,4,5,6};
	print_array(a, 5);
	reverse_array(a,0, 4);;
	print_array(a, 5);
	return 0;

}